tests/test_mi.py

```diff
@@ -245,18 +265,40 @@ def test_machine_learning_error_should_be_normalized(self):
 })
 class ServiceFabricTestCase(ClientTestCase):
 
-    def _test_happy_path(self, app):
+    def _test_happy_path(self, app, *, claims_challenge=None) -> callable:
         expires_in = 1234
         with patch.object(app._http_client, "get", return_value=MinimalResponse(
             status_code=200,
             text='{"access_token": "AT", "expires_on": %s, "resource": "R", "token_type": "Bearer"}' % (
                 int(time.time()) + expires_in),
         )) as mocked_method:
             super(ServiceFabricTestCase, self)._test_happy_path(
-                app, mocked_method, expires_in)
+                app, mocked_method, expires_in, claims_challenge=claims_challenge)
+            return mocked_method
 
-    def test_happy_path(self):
-        self._test_happy_path(self.app)
+    def test_happy_path_with_client_capabilities_should_relay_capabilities(self):
+        self._test_happy_path(self._build_app(client_capabilities=["foo", "bar"])).assert_called_with(
+            'http://localhost',
+            params={
+                'api-version': '2019-07-01-preview',
+                'resource': 'R',
+                'token_sha256_to_refresh': hashlib.sha256(b"AT").hexdigest(),
+                "xms_cc": "foo,bar",
+            },
+            headers={'Secret': 'foo'},
+        )
+
+    def test_happy_path_with_claim_challenge_should_send_sha256_to_provider(self):
+        claims_challenge='{"access_token": {"nbf": {"essential": true, "value": "1563308371"}}}'
+        self._test_happy_path(self.app, claims_challenge=claims_challenge).assert_called_with(
+            'http://localhost',
+            params={
+                'api-version': '2019-07-01-preview',
+                'resource': 'R',
+                'token_sha256_to_refresh': hashlib.sha256(b"AT").hexdigest(),
+            },
```

Comment:
do we need a test of claims + forceRefresh?

Reply:
Not currently applicable, as MSAL Python does not support forceRefresh. We may revisit this when the circumstances changes in the future.